BLUI-251: As Ceviche, I want to know how to build an app store icon (#607)
brightlayer-ui/doc-it: src/docs/style/logo.mdx

-   [PDF (for printing)](https://raw.githubusercontent.com/brightlayer-ui/doc-it/master/src/app/assets/styles/logos/App%20Logos%20Printable.pdf)
-   [SVG](https://github.com/brightlayer-ui/doc-it/tree/master/src/app/assets/styles/logos/templates/SVG)
-   Our [Figma Component Sticker Sheet](https://www.figma.com/community/file/1024360297793425107) also includes a component called "App Logo Template"

## Glyph

A glyph is a simple shape representing an object, action, or idea. Your glyph must be simple, recognizable, and scalable to ensure that your app can be easily identified on multiple devices with different resolutions.

We recommend you select your base glyph from our [icon library](/style/icon-library).

<ImageGrid
    images={[Glyph1, Glyph2]}
    captionsUnderImages={[
        `AVOID: Avoid the use of mnemonic logos or logotypes when possible. They are often not meaningful for first-time users, and can be hard to see in small logo sizes.`,
        `DO: Use a graphical way to represent an application or product.`,
    ]}
/>

> Do not use the Eaton logo as a glyph — your app should not be presented as the "Eaton" app. It is not unusual for a customer to have multiple Eaton products and applications installed on their devices, and if every one of those apps used the Eaton logo it becomes very difficult for a user to tell, at a glance, which application they are looking for.

Reviewer: ```suggestion
Do not use the Eaton logo as a glyph — your app should not be presented as the "Eaton" app. It is not unusual for a customer to have multiple Eaton products and applications installed on their devices — if every one of those apps used the Eaton logo it would be very difficult for a user to differentiate them.
```

<ImageGrid images={[Glyph3]} captionsUnderImages={[`DONT: Do not use the Eaton logo as glyphs.`]} />

You should also verify that your product logo is distinguished enough from other existing Eaton product logos.

Glyphs are usually aligned to the keyline (visible in the template) to avoid bleeding beyond the safe zone.

<ImageGrid images={[GlyphKeyline]} captionsUnderImages={[`Align glyphs to the icon template keyline.`]} regularWidth />

Once the base glyph is decided, you can be creative and add variations to the glyph to make it look more designed.

<ImageGrid
    images={[PolishGlyph1, PolishGlyph2]}
    captionsUnderImages={[

>         `Smart Energy Manager modified the glyph shape to accommodate for a bolt glyph inside.`,

Reviewer: ```suggestion
        `Smart Energy Manager modified the glyph shape to add a bolt glyph inside.`,
```

>         `Predict Pulse modified the glyph size, opacity, and image gradient.`,

Reviewer: Just a small point...since we moved the glyph section to be first, talking about the customization of the background is a tiny bit premature at this point in the doc...I think it's fine, just pointing it out.

    ]}
/>

## Background

Most Eaton app logos use [the primary blue](/style/color?category=ui&name=blue&weight=500) as the base background color. Some also select the background color from one of the [branding colors](/style/color#branding-colors) to stay consistent with colors in their product marketing.

<ImageGrid
    images={[Background1]}
    captionsUnderImages={[`Use the primary blue or one of the branding colors as a foundation for the background.`]}
    regularWidth
/>

<ImageGrid
    images={[Background2, Background3]}
    captionsUnderImages={[
        `DONT: Do not use a photo in the logo.`,
        `DONT: Do not use a screenshot or app UI in the logo.`,
    ]}
/>

<ImageGrid
    images={[BackgroundGradient1, BackgroundGradient2, BackgroundGradient3]}
    captionsUnderImages={[

>         `DO: Most Eaton apps choose to add a subtle corner linear gradient to their app logo background.`,

Reviewer: ```suggestion
        `DO: Most Eaton apps choose to add a subtle linear gradient to corner of their app logo background.`,
```

Author: not technically true... that gradient spans the whole background

Reviewer: 'corner linear gradient' sounds weird to me. How about 'a diagonal gradient'

Author: that is exactly what I ended up doing :D

        `Apps may also consider using a centered radial gradient, especially when the glyph has a round shape.`,
        `A monochromatic background is acceptable.`,
    ]}
    gridImageProps={{ sm: 6, md: 4 }}
/>

## Bottom Badge

The bottom badges are not meant to be modified. You are expected to leave the bottom badge as it is in the logo templates.

<ImageGrid
    images={[BottomBadge1, BottomBadge2, BottomBadge3, BottomBadge4, BottomBadge5]}
    captionsUnderImages={[
        `DO: Use the default badge as it is.`,
        `DONT: Don't modify the badge color.`,
        `DONT: Don't modify the badge logo color.`,
        `DONT: Don't modify the badge size or the logo size.`,
        `DONT: Don't modify the badge effects or add shadows.`,
    ]}
    gridImageProps={{ sm: 6, md: 4 }}
/>

## Export

> When exporting to different platforms, ensure that your logo design is following the platform-specific requirement [as listed below](#logo-requirements-by-platforms).

Reviewer: ```suggestion
When exporting for different platforms, ensure that your logo design follows the platform-specific requirements [as listed below](#logo-requirements-by-platforms).
```

> When exporting to specific resolutions it is sometimes helpful to tweak vector shapes so that they fall on exact pixels, a process also known as "pixel hinting". This helps maintain a “crisp” look and avoids the blurriness that may occur when edges are in between pixels.

Reviewer: ```suggestion
When exporting for specific resolutions it is sometimes helpful to tweak vector shapes so that they fall on exact pixels, a process also known as "pixel hinting". This helps maintain a “crisp” look and avoids the blurriness that may occur when edges are in between pixels.
```

<ImageGrid
    images={[PixelHinting1, PixelHinting2, PixelHinting3]}
    captionsUnderImages={[
        `The original app logo design.`,
        `When this design is shrinked to a small size, the details become illegible.`,
        `Pixel-hinting helps sharpening the app logo at low resolutions.`,
    ]}
    caption={`Pixel hinting is helpful for app logos at low resolutions.`}
    gridImageProps={{ sm: 6, md: 4 }}
/>

> Do not export rounded corners and drop shadows to your final design assets. You are encouraged to preview the logo design with these effects enabled, but your final exported assets should be square-shaped.

Reviewer: ```suggestion
Do not export your final app logo with rounded corners or drop shadows. You are encouraged to preview the logo design with these effects enabled, but your final exported assets should be square-shaped. The additional effects are automatically applied by the application platform.
```

        `3: Favicon for web apps, previewed at a high resolution.`,
        `4: Amazon Alexa Skills' rendering.`,
        `5 – 8: Android's adaptive icon allows phone manufacturers to render app logos in different shapes.`,
    ]}
    regularWidth
/>

Different platforms have different product logo requirements — they often have different shapes, resolutions, safe areas, and parallax layers. Click on the links below to learn more about logo guidelines on each platform.

-   [Apple (iOS, iPadOS, macOS, tvOS, watchOS)](https://developer.apple.com/design/human-interface-guidelines/foundations/app-icons/)
-   Android
    -   [Adaptive Icons](https://developer.android.com/develop/ui/views/launch/icon_design_adaptive)
        -   [Image Asset Studio](https://developer.android.com/studio/write/image-asset-studio.html) in Android Studio
        -   Adaptive Icon [Design Guide](https://medium.com/google-design/designing-adaptive-icons-515af294c783)
    -   [Google Play](https://developer.android.com/distribute/google-play/resources/icon-design-specifications)
-   [Windows](https://learn.microsoft.com/en-us/windows/apps/design/style/iconography/app-icon-design)

## Favicons

> Favicons are product logos for web apps. They can appear in browser tabs, task bars, bookmarks, and phone screens just like native mobile apps, under various image sizes. We highly recommend redesigning the app logo specifically for the favicon to ensure that it shows well at all resolutions.

Reviewer: ```suggestion
Favicons are product logos for web apps. They can appear at different sizes in browser tabs, task bars, bookmarks, and on phone screens (like native mobile apps). We highly recommend redesigning the app logo specifically for the favicon to ensure that it shows well at all resolutions.
```

According to [Mailchimp](https://mailchimp.com/resources/favicon-size/), the optimal favicon dimensions are as follows, in pixels:

-   Browser tab favicons - 16×16
-   Taskbar shortcut icons - 32×32
-   Desktop shortcut icons - 96×96
-   Apple touch icons - 180×180

> Developers should [include multiple favicons](https://blog.hubspot.com/website/what-is-a-favicon) for these different resolutions in the HTML head to accommodate for different favicon usage.

Reviewer: ```suggestion
Developers should [include multiple favicons](https://blog.hubspot.com/website/what-is-a-favicon) for these different resolutions in the HTML head to accommodate these various use cases.
```

Favicons used in browser tabs and taskbar shortcuts resemble the original product logo design, but with the badge removed. It always has the background color to ensure that the glyph can be seen clearly.

<ImageGrid
    images={[Favicon]}
    caption={[
        `Favicon needs to be recognizable at a size as small as 16×16 pixels.`,
        `1: The original product logo, with the bottom badge removed to allow more room for the glyph.`,
        `2: The same logo, rendered at 16×16, a favicon size used in browser tabs.`,
        `3: Optionally, you can consider doing "pixel-hinting" — manually editing the favicon to ensure that it renders well at low resolution.`,
        `4: The favicon in a browser tab.`,
    ]}
    regularWidth
/>

Desktop shortcuts and Apple touch icons should resemble the design of standard app logos.

# Design Specifications

> The templates should already have everything you need. However, if you want to do something fancy, here's the design spec for a default logo:

Reviewer: ```suggestion
The provided templates have implemented these specifications for you already, but they are presented here for your reference if you need to make any customizations:
```

Author: thanks! I definitely needed this one... I really don't know how to express it politely, but you nailed it 😂

<ImageGrid
    images={[Spec, SpecCircle, SpecColor]}
    captionsUnderImages={[
        `A regular app logo.`,

>         `For platforms using round app logo shapes, such as Android's adaptive icon and Amazon Alexa Skills, the Eaton logo in the badge is shrunk to remain visible.`,

Reviewer: Does this mean designers need to design two icons if they want to support iOS and Android? This seems to conflict with the recommendation to use tools to automatically generate your icon formats for you (might need some more clarity around this).

Author: Yes, that is correct. After the base design is done, you plug the design into the android version
